deployer/steps.go

```diff
@@ -7,35 +7,56 @@ import (
 	"os"
 	"path/filepath"
 	"strconv"
+	"time"
 
 	"github.com/kairos-io/AuroraBoot/pkg/ops"
 	"github.com/spectrocloud-labs/herd"
 )
 
+type callback func(context.Context) error
+
+// logStepTime logs the start and end time of a step
+// and the duration it took to complete
+func logStepTime(stepName string, stepFunc callback) callback {
+	return func(ctx context.Context) error {
+		start := time.Now()
+		internal.Log.Logger.Trace().Str("step", stepName).Time("start", start).Msg("Starting step")
+		err := stepFunc(ctx)
+		end := time.Now()
+		duration := end.Sub(start)
+		if err != nil {
+			internal.Log.Logger.Trace().Err(err).Str("step", stepName).Time("start", start).Time("end", end).Str("duration", duration.String()).Msg("Step failed")
+		} else {
+			internal.Log.Logger.Trace().Str("step", stepName).Time("start", start).Time("end", end).Str("duration", duration.String()).Msg("Completed step")
+		}
+		return err
+	}
+}
+
 func (d *Deployer) StepPrepNetbootDir() error {
 	return d.Add(opPrepareNetboot, herd.WithCallback(
-		func(ctx context.Context) error {
+		logStepTime(opPrepareNetboot, func(ctx context.Context) error {
```

Comment:
If we are going to do it in every case, maybe we can make it part of `Add` or `WithCallback`? I haven't checked the code to see but I assume it's possible to just call it once in one of those 2. If not, ignore me.

Reply:
well I guess we could make it part of herd directly and have it as an option to track the step time and report it as part of the step completition or something, like the run or error values on each step

Reply:
If you think it's better...your call.